openmdao/core/driver.py

```diff
@@ -1098,5 +1098,16 @@ def record_iteration(requester, prob, case_name):
             totals = requester.compute_totals(return_format='flat_dict_structured_key')
             data['totals'] = totals
 
+        # Record solver info
+        if requester.recording_options['record_abs_error'] or \
+                requester.recording_options['record_rel_error']:
+            solver = model.nonlinear_solver
```

Comment:
I think you could move this to the 3rd `if` statement.. looks like it is only used there...